plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt

```diff
@@ -252,26 +263,40 @@ class ScaleBarImpl : ScaleBar, View {
     for (i in 0 until pair.second) {
       barPaint.color = if (i % 2 == 0) settings.primaryColor else settings.secondaryColor
       val distanceText = getDistanceText(unitDistance * i)
+      // Make the first text shift to right with borderWidth, the most right text shift to left with INTERNAL_PADDING
+      val xPositionShitForText =
+        when (i) {
+          0 -> {
+            settings.borderWidth
+          }
+          pair.second - 1 -> {
+            -INTERNAL_PADDING.toFloat()
+          }
+          else -> {
+            0f
+          }
+        }
+
       textPaint.getTextPath(
-        distanceText, 0, distanceText.length, settings.marginLeft + unitBarWidth * i,
-        settings.textSize + settings.marginTop, path
+        distanceText, 0, distanceText.length, (unitBarWidth * i) + xPositionShitForText,
+        settings.textSize, path
       )
       if (settings.showTextBorder) {
         canvas.drawPath(path, strokePaint)
       }
       canvas.drawPath(path, textPaint)
       canvas.drawRect(
-        settings.marginLeft + unitBarWidth * i,
-        settings.textBarMargin + settings.textSize + settings.marginTop,
-        settings.marginLeft + unitBarWidth * (1 + i),
-        settings.textBarMargin + settings.textSize + settings.marginTop + settings.height,
+        (settings.borderWidth * 2) + (unitBarWidth * i),
+        settings.textBarMargin + settings.textSize,
+        unitBarWidth * (1 + i),
+        settings.textBarMargin + settings.textSize + settings.height,
```

Comment:
I think most of this parameters can be precomputed in advance isn’t it?

Reply:
Some of them could, but I would rather keep them here because they are very simple calculations.